src/cartesian/CartesianGrid.tsx

```diff
@@ -285,129 +290,178 @@ function VerticalStripes(props: Props) {
+
+    const generatorResult = horizontalCoordinatesGenerator(
+      {
+        yAxis: yAxis
+          ? {
+              ...yAxis,
+              ticks: isHorizontalValues ? horizontalValues : yAxis.ticks,
+            }
+          : undefined,
+        width: chartWidth,
+        height: chartHeight,
+        offset,
+      },
+      isHorizontalValues ? true : syncWithTicks,
+    );
+    warn(
+      Array.isArray(generatorResult),
+      `horizontalCoordinatesGenerator should return Array but instead it returned [${typeof generatorResult}]`,
+    );
+    if (Array.isArray(generatorResult)) {
+      horizontalPoints = generatorResult;
     }
+  }
 
-    return (
-      <g className="recharts-cartesian-grid">
-        <Background
-          fill={this.props.fill}
-          fillOpacity={this.props.fillOpacity}
-          x={this.props.x}
-          y={this.props.y}
-          width={this.props.width}
-          height={this.props.height}
-        />
-        <HorizontalGridLines {...this.props} horizontalPoints={horizontalPoints} />
-
-        <VerticalGridLines {...this.props} verticalPoints={verticalPoints} />
-
-        <HorizontalStripes {...this.props} horizontalPoints={horizontalPoints} />
-
-        <VerticalStripes {...this.props} verticalPoints={verticalPoints} />
-      </g>
+  // No vertical points are specified
+  if ((!verticalPoints || !verticalPoints.length) && isFunction(verticalCoordinatesGenerator)) {
+    const isVerticalValues = verticalValues && verticalValues.length;
+    const generatorResult = verticalCoordinatesGenerator(
+      {
+        xAxis: xAxis
+          ? {
+              ...xAxis,
+              ticks: isVerticalValues ? verticalValues : xAxis.ticks,
+            }
+          : undefined,
+        width: chartWidth,
+        height: chartHeight,
+        offset,
+      },
+      isVerticalValues ? true : syncWithTicks,
     );
+    warn(
+      Array.isArray(generatorResult),
+      `verticalCoordinatesGenerator should return Array but instead it returned [${typeof generatorResult}]`,
+    );
+    if (Array.isArray(generatorResult)) {
+      verticalPoints = generatorResult;
+    }
   }
+
+  return (
+    <g className="recharts-cartesian-grid">
+      <Background
+        fill={propsIncludingDefaults.fill}
+        fillOpacity={propsIncludingDefaults.fillOpacity}
+        x={propsIncludingDefaults.x}
+        y={propsIncludingDefaults.y}
+        width={propsIncludingDefaults.width}
+        height={propsIncludingDefaults.height}
+      />
+      <HorizontalGridLines {...propsIncludingDefaults} horizontalPoints={horizontalPoints} />
+
+      <VerticalGridLines {...propsIncludingDefaults} verticalPoints={verticalPoints} />
+
+      <HorizontalStripes {...propsIncludingDefaults} horizontalPoints={horizontalPoints} />
+
+      <VerticalStripes {...propsIncludingDefaults} verticalPoints={verticalPoints} />
+    </g>
+  );
 }
+
+CartesianGrid.displayName = 'CartesianGrid';
```

Comment:
we can't do this because of https://github.com/recharts/recharts/issues/3615 :(

Reply:
The linked issue says defaultProps but code says displayName, is that related?

I have removed defaultProps static property from this component completely - as far as I can tell it's never used. (Unlike Tooltip.defaultProps).

Reply:
lol sorry, I'm doing too much context switching today. Yeah keeping the display name should be fine.

Is CartesianGrid ever found by any of the `findByType` utils? That is a case where default params will not work

Reply:
Doesn't look like it 👍🏼 https://github.com/search?q=repo%3Arecharts%2Frecharts+ByType&type=code&p=1